### ISSUE TITLE
WebSockets - authentication error on connect does not emit error event.

### ISSUE THREAD

**Issue by mckn** [mqttjs/MQTT.js#448]
Hello,

To start, thanks for a great client!

I found a small issue, if I try to connect via ws to a mqtt server with invalid credentials the client does not emit the error event as it does if I do the same without ws.

Could this be fixed in any way? Would be nice with the same behaviour in both cases.

Best regards,
Marcus

**Comment by mcollina** [comment]
Hi @mckn,

Sorry for not noticing this issue before. Probably it's a bug, either in the server you are using or in MQTT.js. Can you please add a script to reproduce?

Cheers,

Matteo

**Comment by japita-se** [comment]
I confirm this.
I am using browserified version of MQTT.

`<script src="assets/js/browserMqtt.js"></script>`

I am using the authenticate and authorize hooks, as in the WiKi.
With invalid credentials, the server behaves correctly, but no information are returned back  to the client.
For instance, in the web page:

```
client.on('connect',function(res){  
//res is always null. 
})
```

 Same for other methods (subscribe).
In other words, the client has no way to understand what happened upon connecting/subscribing.

**Comment by mcollina** [comment]
Can you please upload a full example? It's really hard to debug in this way.

**Comment by DirkHeinke** [comment]
I had the same problem with mosquitto 1.4.14 but it is fixed with 1.4.15

**Comment by YoDaMa** [comment]
@DirkHeinke said this problem was resolved for him plus there's no repro possible so we will close this issue.
